[crashrpt] Add new port

tbdrake opened the pull request:
This builds the CrashRpt and CrashRptProbe libraries and CrashSender executable that must be distributed with apps that use this package. This is my first attempt at creating a port and I have some notes and questions, so I created this as a draft pull request.

### Vendored dependencies
The CrashRpt repository includes old vendored dependencies for the following libraries. These have been replaced (except minizip) in this port with vcpkg dependencies so that these dependencies can be updated and maintained according to the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#do-not-use-vendored-dependencies).

* zlib
* ~minizip~ (not replaced since CrashRpt's copy has modifications that are necessary for CrashRpt's functionality)
* libpng
* libjpeg (replaced with libjpeg-turbo)
* tinyxml
* libogg
* libtheora
* wtl
* dbghelp (replaced with new port that copies from Windows 10 SDK in build environment)

### CrashSender exe and dependencies copied to "tools" directory
The portfile here copies the CrashSender executables to the "tools" and "debug/tools" directories. CrashSender also depends on a .ini file with localized strings, so the .ini files from the CrashRpt repo are also copied to the "tools" directory. Is there a more appropriate standard location besides "tools" for these files?

### Distributing dbghelp.dll
Notes about redistributing dbghelp.dll
https://docs.microsoft.com/en-us/windows/win32/debug/dbghelp-versions?redirectedfrom=MSDN
> The most current versions of DbgHelp.dll, SymSrv.dll, and SrcSrv.dll are available as a part of the Debugging Tools For Windows package. The redistribution policies for these included DLLs were specifically designed to make it as easy as possible for people to include these files in their own packages and releases.

https://docs.microsoft.com/en-us/windows/win32/debug/calling-the-dbghelp-library
> When using DbgHelp, the best strategy is to install a copy of the library from the Debugging Tools For Windows package in the application directory logically adjacent to the software that calls it.

The CrashRpt documentation has this note:
> Note:
> dbghelp.dll library is required to generate minidump files. dbghelp.dll is included in Windows XP distribution and in later versions of Windows, and most user machines have this library installed. This means it is possible (but not recommended) to ignore distribution of dbghelp.dll, if you strongly wish.
> 

### Usage info
In a test app that uses the crashrpt package I added a post-build event like this to copy (and rename the .ini file) to the target directory like this:
```
copy "$(VcpkgCurrentInstalledDir)tools\crashrpt\crashrpt_lang_EN.ini" "$(TargetDir)crashrpt_lang.ini"
copy "$(VcpkgCurrentInstalledDir)tools\crashrpt\CrashSender1403.exe" "$(TargetDir)"
```
https://github.com/tbdrake/vcpkg-samples/tree/master/crashrpt/CrashRptSample

resolves #560

draketb committed:
Add port files for crashrpt

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/not_signed)](https://cla.opensource.microsoft.com/microsoft/vcpkg?pullRequest=9162) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/vcpkg?pullRequest=9162) before we can accept your contribution.<br/><br/>:x: tbdrake [sign now](https://cla.opensource.microsoft.com/microsoft/vcpkg?pullRequest=9162)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/microsoft/vcpkg?pullRequest=9162) it.</sub>

draketb committed:
Fix x64 output paths and fail port install if architecture is not x86 or x64

JackBoosY commented on ports/crashrpt/portfile.cmake:
```diff
@@ -0,0 +1,68 @@
+include(vcpkg_common_functions)
```
We don't need to include it in `portfile.cmake` anymore, please remove this line.

JackBoosY commented on ports/crashrpt/portfile.cmake:
```diff
@@ -0,0 +1,68 @@
+include(vcpkg_common_functions)
+
+if(NOT VCPKG_TARGET_IS_WINDOWS)
```
Please use `vcpkg_fail_port_install(ON_TARGET "OSX" "Linux")` instead.

tbdrake replied:
There are a few other targets that this port wouldn't support, should those also be list like this?
 `vcpkg_fail_port_install(ON_TARGET "OSX" "Linux" "UWP" "Android" "FreeBSD")`

Here is the documentation where I found the list of targets:
https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/vcpkg_fail_port_install.md#on_target
> targets for which the build should fail early. Valid targets are from VCPKG_IS_TARGET_ (see vcpkg_common_definitions.cmake)

https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/vcpkg_common_definitions.md#the-following-variables-are-available
> VCPKG_TARGET_IS_\<target\>                 with \<target\> being one of the following: WINDOWS, UWP, LINUX, OSX, ANDROID, FREEBSD. only defined if \<target\>

JackBoosY replied:
@tbdrake `Android` and `FreeBSD` are part of `Linux`, so we can just use `Linux` here.
If this port does not support UWP, you should add `UWP`.

JackBoosY commented on ports/crashrpt/portfile.cmake:
```diff
@@ -0,0 +1,68 @@
+        -DCRASHRPT_LINK_CRT_AS_DLL=${CRASHRPT_LINK_CRT_AS_DLL}
+)
+
+# CrashRpt does not have an install target, so build the two targets that are needed and copy the output
```
We prefer to use patches to fix installation issues in the source code.

tbdrake replied:
Updated today to apply a patch to add a CMake install target and call `vcpkg_install_cmake()` from portfile.cmake

draketb committed:
Use vcpkg_fail_port_install

LilyWangL commented:
Pinging @tbdrake for response. Is work still being done for this PR?

tbdrake commented:
Sorry I have not spent any more time on this recently, but still would like
to. I have a week of vacation from my day job next week and plan to spend
some time on this then.

On Tue, Apr 14, 2020, 4:42 AM Lily <notifications@github.com> wrote:

> Pinging @tbdrake <https://github.com/tbdrake> for response. Is work still
> being done for this PR?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/microsoft/vcpkg/pull/9162#issuecomment-613307638>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AETGDNPFUFII3QH45DTIL2DRMQOY3ANCNFSM4JTHSOUA>
> .
>

draketb committed:
Merge branch 'master' into crashrpt

draketb committed:
Add patch for CMakeLists.txt add install target for CrashRpt and CrashSender

tbdrake commented:
I'm still planning to look into replacing the vendored dependencies.

johannesstricker commented:
I'm also interested in this and would be willing to help, but I couldn't find any information on how to list dependencies for a new port. Is it just a `Build-Depends: ...` in the `CONTROL` file?

draketb committed:
Replace vendored dependencies for minizip, png, and zlib with packages from vcpkg

draketb committed:
Fix 002-find-minizip-png-zlib.patch, accidentally included folder move/renaming in previous commit

draketb committed:
Add vcpkg_copy_tool_dependencies so that dlls for CrashSender .exe will be copied to tools folder

draketb committed:
Add install for crprober.exe tool

draketb committed:
Replace libogg and libtheora vendored dependencies using find_library() and find_path()

draketb committed:
Replace vendored tinyxml library

draketb committed:
Replace vendored wtl library

draketb committed:
Replace vendored JPEG library with libjpeg-turbo from vcpkg

JackBoosY commented:
/azp run

draketb committed:
Add patch to install demos and test applications to tools directory and add vcpkg port features to enable building the tests and demos

draketb committed:
Add port of dbghelp that copies files from Windows 10 SDK. Use this crashrpt to replace vendored dependency of dbghelp.

tbdrake commented:
@johannesstricker Thank you, any help and review would be appreciated. I worked on replacing the vendored dependencies as best I could by looking at existing ports in vcpkg. Some vcpkg documentation was also helpful:
*  https://github.com/microsoft/vcpkg/blob/master/docs/examples/installing-and-using-packages.md#handling-libraries-without-native-cmake-support
* https://github.com/microsoft/vcpkg/blob/master/docs/users/integration.md#cmake-toolchain-file-recommended-for-open-source-cmake-projects

Besides the `Build-Depends` in the `CONTROL` file I also had to update some of the `CMakeLists.txt` to use `find_package()/find_library()/find_path()` to locate and use the vcpkg libraries rather than the versions in the CrashRpt git repo. I also made a few small code changes in the patch files in this branch that were required to fix compilation issues when using the versions of the dependencies from vcpkg (newer than the ones CrashRpt was originally written with).

I added feature options `tests` and `demos` to build the test and demo applications (console demo, WTL demo, MFC demo) that are included with CrashRpt, but some tests are failing unexpectedly on my machine and I have not verified that the demos are working properly. Any help in testing and fixing issues so that the tests pass and demos work properly would be greatly appreciated. I have not had time to look in to the causes of the failures and if these are new failures because of the updated dependencies.

One issue I noticed for example is that the "Export..." button in Error Report Details is not successfully saving the error report zip file (it seems to silently fail when I click "Save"):  
![image](https://user-images.githubusercontent.com/19292597/80318936-9247fb80-87db-11ea-838d-099403efddea.png)

JackBoosY commented:
Does you need my help?

tbdrake commented:
@JackBoosY 
Yes I could use help. One thing I am not sure how to handle correctly is the CI pipeline, particularly for the dbghelp port I added here which crashrpt depends on.

The build for dbghelp:x86-windows [here](https://github.com/microsoft/vcpkg/pull/9162/checks?check_run_id=620318873) failed and stdout-x86-windows-log shows this message that dbghelp.lib could not be found where expected:
```
CMake Error at ports/dbghelp/portfile.cmake:16 (message):
  Cannot find Windows 10.0.18362.0 SDK.  File does not exist: C:\Program
  Files (x86)\Windows Kits\10\Debuggers\lib\x86\dbghelp.lib
Call Stack (most recent call first):
  scripts/ports.cmake:90 (include)
```

I think this is because the dbghelp library is included in the Windows SDK installation only if _Debugging Tools for Windows_ feature is included, so maybe the CI system does not have this installed:
![image](https://user-images.githubusercontent.com/19292597/81484585-8e24d080-9214-11ea-986b-4c77a9be1b40.png)

How should I fix this?

Commit:
[dbghelp] Improve portfile.cmake

Commit:
Update baseline

Commit:
update CONTROL file

Commit:
Merge branch 'master' of https://github.com/microsoft/vcpkg into crashrpt

JackBoosY commented:
@BillyONeal This PR requires the pipeline machine to install the Windows SDK with debugging tools.

BillyONeal commented:
> @BillyONeal This PR requires the pipeline machine to install the Windows SDK with debugging tools.

I don't think we can test it then, unless we have a reliable way to install that unattended.

JackBoosY commented:
/azp run

JackBoosY commented:
@tbdrake Please resolve file conflicts first.

Thanks.

draketb committed:
Merge remote-tracking branch 'upstream/master' into crashrpt

# Conflicts:
#	scripts/ci.baseline.txt

tbdrake commented:
@JackBoosY Thank you for helping, and merge conflict should now be resolved, it was just in scripts/ci.baseline.txt:

JackBoosY commented:
@tbdrake According to the disscussion, please add the following codes to _VCPKG_PATH/scripts/ci.baseline.txt_:
```
# Since pipeline cannot automatically install dbghelp dependency, skip this detection
dbghelp:arm64-windows=skip
dbghelp:arm-uwp=skip
dbghelp:x64-uwp=skip
dbghelp:x64-windows=skip
dbghelp:x64-windows-static=skip
dbghelp:x86-windows=skip
```

draketb committed:
Merge branch 'master' of https://github.com/microsoft/vcpkg into crashrpt

draketb committed:
Update ci.baseline.txt to skip dbghelp on Windows platforms

JackBoosY commented:
@tbdrake Does this PR ready for review?

tbdrake commented:
> @JackBoosY Sorry for the delay. I think this is close but not quite ready. I think that updating the dependencies may have caused some of the unit tests in crashrpt[tests] to fail (mentioned in [this comment](https://github.com/microsoft/vcpkg/pull/9162#issuecomment-619621605)), but I'm not sure yet. I plan to look into that on 6/15/2020, and possibly may add more patches to this port if that is required to make the tests pass.

I looked into this some today but still would like to look into these issues more before making this pull request ready for review. 23 of 68 tests fail in the Testsd.exe built by this port, but only 2 of 68 fail when I build crashrpt as-is from its sourceforge repo using its vendored dependencies (not using vcpkg).

[crashrpt-testsd-out.txt](https://github.com/microsoft/vcpkg/files/4783480/crashrpt-testsd-out.txt)
[vcpkg-crashrpt-testsd-out.txt](https://github.com/microsoft/vcpkg/files/4783481/vcpkg-crashrpt-testsd-out.txt)

JackBoosY commented:
@tbdrake Can you provide the way to reproduce this?

Thanks.

tbdrake commented:
@JackBoosY Here are the steps I'm doing to build and run the CrashRpt automated tests in Release. Results from my earlier comment were from Debug builds and the steps are similar and I get different results using Debug as well. Also note that there are 3 `DeliveryTests` that are expected to fail unless you have the additional setup required (there's a note [here](http://crashrpt.sourceforge.net/docs/html/automated_tests.html)).

**Using vcpkg and this port:**
1. Clone this pull request's branch: `git clone https://github.com/tbdrake/vcpkg.git -b crashrpt`
2. In vcpkg directory cloned above, build vcpkg: `bootstrap-vcpkg.bat`
3. Install crashrpt including the `tests` feature: `vcpkg.exe install crashrpt[tests]`
4. Run the crashrpt automated tests: `installed/x86-windows/tools/crashrpt/Tests.exe`
5. Enter `n` for first prompt about testing UNICODE compatibility, then enter a blank line in the second prompt to run all tests. Here is the test summary I get:
```
=== Summary ===

Error list:
1: In test: CrashRptAPITests::Test_crInstallW_zero_info Expr: nInstallResult==0
2: In test: CrashRptAPITests::Test_crInstallA_zero_info Expr: nInstallResult==0
3: In test: CrashRptAPITests::Test_crInstallA_twice Expr: nInstallResult==0
4: In test: CrashRptAPITests::Test_crInstallA_short_path_name Expr: nInstallResult==0
5: In test: CrashRptAPITests::Test_crInstallW_short_path_name Expr: nInstallResult==0
6: In test: CrashRptAPITests::Test_crInstallToCurrentThread2 Expr: nInstResult==0
7: In test: CrashRptAPITests::Test_crInstallToCurrentThread2_concurrent Expr: nInstResult==0
8: In test: CrashRptAPITests::Test_crAddFile2A Expr: nInstallResult==0
9: In test: CrashRptAPITests::Test_crAddFile2W Expr: nInstallResult==0
10: In test: CrashRptAPITests::Test_crAddScreenshot Expr: nInstallResult==0
11: In test: CrashRptAPITests::Test_crAddScreenshot2 Expr: nInstallResult==0
12: In test: CrashRptAPITests::Test_crAddPropertyA Expr: nInstallResult==0
13: In test: CrashRptAPITests::Test_crAddPropertyW Expr: nInstallResult==0
14: In test: CrashRptAPITests::Test_crAddRegKeyA Expr: nInstallResult==0
15: In test: CrashRptAPITests::Test_crAddRegKeyW Expr: nInstallResult==0
16: In test: CrashRptAPITests::Test_crAddVideo Expr: nInstallResult==0
17: In test: CrashRptAPITests::Test_crAddVideo_defaults Expr: nInstallResult==0
18: In test: CrashRptAPITests::Test_crAddVideo_crash Expr: nInstallResult==0
19: In test: CrashRptAPITests::Test_crSetCrashCallbackA Expr: nInstall==0
20: In test: CrashRptAPITests::Test_crSetCrashCallbackW Expr: nInstall==0
21: In test: CrashRptAPITests::Test_crSetCrashCallbackW_stage Expr: nInstall==0
22: In test: CrashRptAPITests::Test_crSetCrashCallbackW_cancel Expr: nInstall==0
23: In test: CrashRptAPITests::Test_crGenerateErrorReport Expr: nInstResult==0
24: In test: CrashRptAPITests::Test_crGetLastErrorMsgA Expr: nInstallResult==0
25: In test: CrashRptAPITests::Test_crGetLastErrorMsgW Expr: nInstallResult==0
26: In test: CrashRptAPITests::Test_CrAutoInstallHelper Expr: cr_install_helper.m_nInstallStatus==0
27: In test: CrashRptAPITests::Test_CrThreadAutoInstallHelper Expr: cr_install_helper.m_nInstallStatus==0
28: In test: CrashRptAPITests::Test_crInstall_in_different_folder Expr: bCopy3
29: In test: CrashRptAPITests::Test_symbol_file_exists Expr: dwAttrs!=INVALID_FILE_ATTRIBUTES Msg: File does not exist: c:\Users\tdrake\Desktop\vcpkg\installed\x86-windows\tools\crashrpt\CrashRpt1403.pdb
30: In test: CrashRptProbeAPITests::SetUp Expr: bCreateReport
31: In test: Test_crpOpenErrorReportA Expr: SetUp Failure
32: In test: Test_crpOpenErrorReportW Expr: SetUp Failure
33: In test: Test_crpCloseErrorReport Expr: SetUp Failure
34: In test: Test_crpExtractFileW Expr: SetUp Failure
35: In test: Test_crpExtractFileA Expr: SetUp Failure
36: In test: Test_crpGetLastErrorW Expr: SetUp Failure
37: In test: Test_crpGetLastErrorA Expr: SetUp Failure
38: In test: Test_crpGetPropertyW Expr: SetUp Failure
39: In test: Test_crpGetPropertyA Expr: SetUp Failure
40: In test: Test_crpGetProperty Expr: SetUp Failure
41: In test: Test_crashrptprobe_dll_file_version Expr: SetUp Failure
42: In test: TearDown Expr: SetUp Failure
43: In test: CrproberTests::SetUp Expr: bCreateReport
44: In test: Test_help Expr: SetUp Failure
45: In test: Test_output Expr: SetUp Failure
46: In test: Test_extract_file Expr: SetUp Failure
47: In test: Test_get Expr: SetUp Failure
48: In test: TearDown Expr: SetUp Failure
49: In test: DeliveryTests::Test_HttpDelivery Expr: nInstResult==0
50: In test: DeliveryTests::Test_SmtpDelivery Expr: nInstResult==0
51: In test: DeliveryTests::Test_SmtpDelivery_proxy Expr: nInstResult==0
52: In test: DeliveryTests::Test_SMAPI_Delivery Expr: nInstResult==0
53: In test: ExceptionHandlerTests::Test_CatchException Expr: bFind Msg: Could not find generated crash report file for exception type: 11 SIGSEGV signal
54: In test: LangFileTests::Test_lang_file_versions Expr: nActualVer==CRASHRPT_VER

 Time elapsed: 19.14 sec.
   Test count: 72
 Tests passed: 18
 Tests failed: 54
```

**Using original CrashRpt git repo, CMake GUI, and Visual Studio:**
1. Clone crashrpt: `git clone https://git.code.sf.net/p/crashrpt/code crashrpt`
2. Open CMake GUI and set the source code and binaries directory to above location of crashrpt repo:
![image](https://user-images.githubusercontent.com/19292597/84959193-df09cd80-b0cc-11ea-914f-c7ee318ac759.png)
3. Click Configure and select Win32 for the Optional platform generator, then Finish:
![image](https://user-images.githubusercontent.com/19292597/84959267-006ab980-b0cd-11ea-8bae-94502fd2dad1.png)
4. Click Generate
5. Click Open Project
6. In Visual Studio, select Release configuration and then Build -> Rebuild Solution
7. Set the Tests project as the startup project, then click the Run button:
![image](https://user-images.githubusercontent.com/19292597/84959452-5dff0600-b0cd-11ea-8412-444c55fbc7c3.png)
8. In Tests.exe window, enter `n` for first prompt about testing UNICODE compatibility, then enter a blank line in the second prompt to run all tests. Here is the test summary I get:
```
=== Summary ===

Error list:
1: In test: CrashRptAPITests::Test_symbol_file_exists Expr: dwAttrs!=INVALID_FILE_ATTRIBUTES Msg: File does not exist: C:\Users\tdrake\Desktop\crashrpt\bin\CrashRpt1403.pdb
2: In test: DeliveryTests::Test_HttpDelivery Expr: dwExitCode==0
3: In test: DeliveryTests::Test_SmtpDelivery Expr: dwExitCode==0
4: In test: DeliveryTests::Test_SmtpDelivery_proxy Expr: dwExitCode==0
5: In test: ExceptionHandlerTests::Test_CatchException Expr: _tcscmp(szBuffer, _T("crEmulateCrash"))==0

 Time elapsed: 44.59 sec.
   Test count: 72
 Tests passed: 67
 Tests failed: 5
```

tbdrake commented:
@JackBoosY I realized today some of those test failures in my previous comment under **Using vcpkg and this port:** are just because there needs to be a `crashrpt_lang.ini` file in the same directory as `CrashSender1403.exe`, so just need to copy/rename one of the lang files before running the `Tests.exe` built by vcpkg:
* `cp ./installed/x86-windows/tools/crashrpt/crashrpt_lang_EN.ini ./installed/x86-windows/tools/crashrpt/crashrpt_lang.ini`

However I still get these test failures: 
```
=== Summary ===

Error list:
1: In test: CrashRptAPITests::Test_crAddFile2A Expr: nResult3==0
2: In test: CrashRptAPITests::Test_crAddFile2W Expr: nResult3==0
3: In test: CrashRptAPITests::Test_symbol_file_exists Expr: dwAttrs!=INVALID_FILE_ATTRIBUTES Msg: File does not exist: C:\Users\tdrake\Desktop\vcpkg\installed\x86-windows\tools\crashrpt\CrashRpt1403.pdb
4: In test: CrashRptProbeAPITests::SetUp Expr: bCreateReport
5: In test: Test_crpOpenErrorReportA Expr: SetUp Failure
6: In test: Test_crpOpenErrorReportW Expr: SetUp Failure
7: In test: Test_crpCloseErrorReport Expr: SetUp Failure
8: In test: Test_crpExtractFileW Expr: SetUp Failure
9: In test: Test_crpExtractFileA Expr: SetUp Failure
10: In test: Test_crpGetLastErrorW Expr: SetUp Failure
11: In test: Test_crpGetLastErrorA Expr: SetUp Failure
12: In test: Test_crpGetPropertyW Expr: SetUp Failure
13: In test: Test_crpGetPropertyA Expr: SetUp Failure
14: In test: Test_crpGetProperty Expr: SetUp Failure
15: In test: Test_crashrptprobe_dll_file_version Expr: SetUp Failure
16: In test: TearDown Expr: SetUp Failure
17: In test: CrproberTests::SetUp Expr: bCreateReport
18: In test: Test_help Expr: SetUp Failure
19: In test: Test_output Expr: SetUp Failure
20: In test: Test_extract_file Expr: SetUp Failure
21: In test: Test_get Expr: SetUp Failure
22: In test: TearDown Expr: SetUp Failure
23: In test: DeliveryTests::Test_SMAPI_Delivery Expr: dwExitCode!=0
24: In test: ExceptionHandlerTests::Test_CatchException Expr: bFind Msg: Could not find generated crash report file for exception type: 11 SIGSEGV signal
25: In test: LangFileTests::Test_lang_file_versions Expr: nActualVer==CRASHRPT_VER

 Time elapsed: 32.06 sec.
   Test count: 72
 Tests passed: 47
 Tests failed: 25
```

JackBoosY commented:
There are indeed many more errors, but I don’t know why.

draketb committed:
* Combined patch files
* Use vendored minizip since it has modifications to support Unicode file paths using wchar_t* which CrashRpt depends on
* Add feature "probe" in order to allow excluding the CrashRptProbe library

tbdrake commented:
I found that the issue causing some of the CrashRpt unit tests to fail was that the copy of minizip in CrashRpt has modifications that are needed for CrashRpt to work properly, so when I had changed it to use minizip from vcpkg those things no longer worked. So in the latest commit today I changed this port back to using CrashRpt's vendored copy of minizip.

One example of a difference in minizip that caused an issue:

https://sourceforge.net/p/crashrpt/code/ci/master/tree/thirdparty/minizip/ioapi.c#l116:
```
file = _wfopen((wchar_t*)filename, mode_fopen);
```
vs https://github.com/madler/zlib/blob/master/contrib/minizip/ioapi.c#L127  (`FOPEN_FUNC` is `fopen64` on Windows)
```
file = FOPEN_FUNC((const char*)filename, mode_fopen);
```

draketb committed:
Merge branch 'master' into crashrpt

# Conflicts:
#	scripts/ci.baseline.txt

JackBoosY commented:
Our pipeline machine has some problems, please wait for fix it.

JackBoosY commented:
/azp run

draketb committed:
Merge branch 'master' into crashrpt

strega-nil commented on ports/dbghelp/portfile.cmake:
```diff
@@ -0,0 +1,29 @@
+
+vcpkg_get_windows_sdk(WINDOWS_SDK)
+
+if (WINDOWS_SDK MATCHES "10.")
```
```suggestion
if (WINDOWS_SDK MATCHES "^10.")
```

or, alternatively,

```suggestion
if (WINDOWS_SDK VERSION_GREATER "10")
```

tbdrake replied:
Thank you for the suggestion, updated to use VERSION_GREATER

draketb committed:
Use VERSION_GREATER instead of MATCHES to compare WINDOWS_SDK version in dbghelp portfile.cmake. Remove redundant check of WINDOWS_SDK version later in script.
